.github/workflows/cleanup-old-workflows.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Cleanup Github Workflows"
+
+on:
+  schedule:
+    - cron: '42 * * * *'
+
+jobs:
+  cleanup-workflows:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Mattraks/delete-workflow-runs@c1a2f0de3fa4134a2357de83c92acff6e298bb8a # pin@v2.0.3
```

Comment:
How does this behave with rate-limiting by the Github API? Does it slow down when encountering HTTP 429s? Might this break other workflows that call the Github API in case different runners share a request rate limit?

Reply:
From the [respective part of the documentation](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#requests-from-github-actions) I read, that there is a limit of 1000 requests per hour for usage of the `GITHUB_TOKEN`.
While there is a way to [inspect the rate limit for access with a _personal_ access token](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#checking-your-rate-limit-status), I think this inspection method does not apply for the `GITHUB_TOKEN` as it is not a personal access token.

That being said towards the general topic of rate limits, I did test this workflow, and had no issue with rate limits whatsoever. There was no impact on other workflows observed during (or after) the test.
It should be mentioned that the execution policy was adopted to minimize the impact on rate limits. I.e. the execution was set to hourly, such that every hour only a view requests are made, and they will not matter with the hourly reset of the quota.